lua/luasnip/nodes/snippet.lua

```diff
@@ -228,9 +245,10 @@ local function init_snippet_context(context, opts)
 	end
 	effective_context.trig_matcher = engine(effective_context.trigger)
 
-	effective_context.condition = context.condition
-		or opts.condition
-		or true_func
+	effective_context.context_matcher = context.context_matcher
```

Comment:
What is your intuition for the name `context_matcher`? It doesn't really mean anything to me :sweat_smile:

Reply:
I think the name is not good enough either. The function returns some properties that are used in snippet expanding. Actually, the `context` refers the context in snippet expanding. And the `matcher` refers if the snippet matches or not.
Do you have any suggestion?

Reply:
I think we should go as generic as possible, because it's not impossible this will be used for even more stuff later on.
The only thing that would define it then is when it is called, so I'd suggest something like `post_trig_match` (although that is very verbose.. just `post_match` maybe? But that would imply that the snippet is matched, and we couldn't change that anymore, which we can..)

Reply:
Maybe `get_expanding_properties`? If this `get` fails, obviously the snippet can't be expanded.

Reply:
Works, but still not a clear winner for me :sweat_smile:
I think it's just that `get` sounds too much like there certainly is something.. Maybe `resolve`, `compute`..
`compute_expand_params` (`expand_params` because that's essentially where this stuff ends up?)

Reply:
But all of those are too long :(((

Reply:
I update the name to `resolve_expand_params`~